### DISCUSSION TITLE
GDB support?

### DISCUSSION THREAD

**Discussion by sleffler** [CHERIoT-Platform/cheriot-rtos#197] [Q&A]
Is there any work on GDB support? I found this repo: https://github.com/CTSRD-CHERI/gdb. Or perhaps this q is better posted to the cheriot-llvm discussions?

**Comment by davidchisnall** [comment]
There is not yet any debugger support, it is something that we'd like to add but have not yet had time to work on.

**Reply by davidchisnall** [reply]
(@sleffler knows most of this, but for other folks coming to this discussion)

We now have lldb working.  SCI open sourced a CHERIoT version of the debug module and we have a fork of OpenOCD that provides the gdbserver stub. Setting this all up is a bit tricky, we’ll try to document all of it soon.

For simulation, the MPact sim that Google contributed runs around 50x faster than the Sail model and has a built-in debugger.
